src/main/java/org/opensearch/security/configuration/ConfigurationRepository.java

```diff
@@ -312,6 +358,75 @@ private void waitForSecurityIndexToBeAtLeastYellow() {
         }
     }
 
+    // viable for testing
+    protected Future<Void> initDefaultConfiguration() {
+        return threadPool.generic().submit(() -> {
```

Comment:
Can `securityIndexHandler.createIndex(...)` or `uploadSecurityConfiguration()` be called without a thread, they already use listeners for async actions?

Reply:
Unfortunately no :-( all operations withing event listener must be called in the thread pool, otherwise it fails.

Reply:
But I will change it. In our case we need to trigger initialization when we already have a manager node.